src/main/java/com/zaxxer/nuprocess/linux/ProcessEpoll.java

```diff
@@ -67,7 +79,17 @@ private ProcessEpoll(int lingerIterations)
          throw new RuntimeException("Unable to create kqueue, errno: " + errno);
       }
 
-      triggeredEvent = new EpollEvent();
+      triggeredEvents = new EpollEvent[EPOLL_MAX_EVENTS];
+
+      EpollEvent.EpollEventPrototype epollEvent = Structure.newInstance(EpollEvent.EpollEventPrototype.class);
+      EpollEvent.EpollEventPrototype[] array = (EpollEvent.EpollEventPrototype[]) epollEvent.toArray(EPOLL_MAX_EVENTS);
+
+      for (int i = 0; i < EPOLL_MAX_EVENTS; i++) {
+         EpollEvent.EpollEventPrototype proto = array[i];
+         triggeredEvents[i] = new EpollEvent(proto.getPointer());
+      }
```

Comment:
I'm not sure I really understand the benefit here. The existing code (without your changes) allocates exactly one `EpollEventPrototype` ever, in the `static` initializer for `EpollEvent`, to determine the layout/size of the structure. Everything past that allocates `EpollEvent` directly--without a prototype.

Here, you've replaced the `EpollEvent` creation/initialization with creation of an array of prototypes, and then you reuse their memory by re-wrapping them as `EpollEvent`. How is that "better" than just using `EpollEvent` directly?
```java
EpollEvent epollEvent = Strucutre.newInstance(EpollEvent.class);
EpollEvent[] array = (EpollEvent[]) epollEvent.toArray(EPOLL_MAX_EVENTS);
```

In both cases, you're only allocating the underlying JNA memory once. Did you think the current code allocated it twice? I suspect your changes here actually yield _more_ overall allocations per `EpollEvent`, when you consider both the native side and the JVM side, because `EpollEventPrototype` has its own `EpollData` nested union it creates, and then you _still_ need to create an `EpollEvent` too.